src/kinetics/ReactionFactory.cpp

```diff
@@ -0,0 +1,245 @@
+    addAlias("pressure-dependent-Arrhenius", "pdep_arrhenius");
+    reg_XML("pressure-dependent-Arrhenius",
+            [](Reaction* R, const XML_Node& node) {
+                setupPlogReaction(*(PlogReaction*)R, node);
+            });
+    reg_AnyMap("pressure-dependent-Arrhenius",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupPlogReaction(*(PlogReaction*)R, node, kin);
+               });
+
+    // register Chebyshev reactions
+    reg("Chebyshev", []() { return new ChebyshevReaction(); });
+    addAlias("Chebyshev", "chebyshev");
+    reg_XML("Chebyshev",
+            [](Reaction* R, const XML_Node& node) {
+                setupChebyshevReaction(*(ChebyshevReaction*)R, node);
+            });
+    reg_AnyMap("Chebyshev",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupChebyshevReaction(*(ChebyshevReaction*)R, node, kin);
+               });
+
+    // register custom Python reactions
+    reg("custom-Python", []() { return new CustomPyReaction(); });
+    reg_XML("custom-Python",
+            [](Reaction* R, const XML_Node& node) {
+                throw CanteraError("ReactionFactory", "Custom Python reactions "
+                                   "cannot be created from XML nodes'");
+            });
+    reg_AnyMap("custom-Python",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupCustomPyReaction(*(CustomPyReaction*)R, node, kin);
+               });
+
+    // register custom Python reactions
+    reg("elementary-new", []() { return new TestReaction(); });
+    reg_XML("elementary-new",
+            [](Reaction* R, const XML_Node& node) {
+                throw CanteraError("ReactionFactory", "Test reactions "
+                                   "cannot be created from XML nodes'");
+            });
+    reg_AnyMap("elementary-new",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupTestReaction(*(TestReaction*)R, node, kin);
+               });
+
+    // register interface reactions
+    reg("interface", []() { return new InterfaceReaction(); });
+    addAlias("interface", "surface");
+    addAlias("interface", "edge");
+    addAlias("interface", "global");
+    reg_XML("interface",
+            [](Reaction* R, const XML_Node& node) {
+                setupInterfaceReaction(*(InterfaceReaction*)R, node);
+            });
+    reg_AnyMap("interface",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupInterfaceReaction(*(InterfaceReaction*)R, node, kin);
+               });
+
+    // register electrochemical reactions
+    reg("electrochemical", []() { return new ElectrochemicalReaction(); });
+    addAlias("electrochemical", "butlervolmer_noactivitycoeffs");
+    addAlias("electrochemical", "butlervolmer");
+    addAlias("electrochemical", "surfaceaffinity");
```

Comment:
I'd really like to avoid dragging some of these dead ends into any new code. These reaction types don't exist in YAML, and I'm pretty sure they never worked correctly even in the XML format.

More broadly, this kind of thing is my main motivation for suggesting that all the XML stuff should just be left as-is. I guess the one concern is whether that gets in the way of converting the existing rate types to use the new, generic formulation or not. It's not clear to me whether we would be able to change the existing classes without a round of deprecations, so if those deprecated classes were still used with XML for Cantera 2.6 and then both XML and the old rate classes go away together in 3.0, it would be fine.

Reply:
No problem removing those aliases - I created those a year ago (prior to the discussion and ensuing deprecation of some of these).